packages/ripple-keypairs/src/index.ts

```diff
@@ -38,74 +42,84 @@ const secp256k1 = {
-        .getPublic()
-        .encodeCompressed(),
-    )
+    const publicKey = bytesToHex(nobleSecp256k1.getPublicKey(derived, true))
     return { privateKey, publicKey }
   },
 
-  sign(message, privateKey): string {
-    return bytesToHex(
-      Secp256k1.sign(hash(message), hexToBytes(privateKey), {
-        canonical: true,
-      }).toDER(),
+  sign(message: ByteArray, privateKey: HexString): string {
+    // Some callers pass the privateKey with the prefix, others without.
+    // @noble/curves will throw if the key is not exactly
+    // 32 bytes, so we normalize it before passing to the sign method.
+    assert.ok(
+      (privateKey.length === 66 && privateKey.startsWith(SECP256K1_PREFIX)) ||
+        privateKey.length === 64,
     )
+    const normed = privateKey.length === 66 ? privateKey.slice(2) : privateKey
+    return nobleSecp256k1
+      .sign(hash(message), normed)
+      .toDERHex(true)
+      .toUpperCase()
   },
 
   verify(message, signature, publicKey): boolean {
-    return Secp256k1.verify(hash(message), signature, hexToBytes(publicKey))
+    const decoded = nobleSecp256k1.Signature.fromDER(signature)
+    return nobleSecp256k1.verify(decoded, hash(message), publicKey)
   },
 }
 
 const ed25519 = {
-  deriveKeypair(entropy: Uint8Array): {
+  deriveKeypair(entropy: ByteArray): {
     privateKey: string
     publicKey: string
   } {
     const prefix = 'ED'
     const rawPrivateKey = hash(entropy)
     const privateKey = prefix + bytesToHex(rawPrivateKey)
     const publicKey =
-      prefix + bytesToHex(Ed25519.keyFromSecret(rawPrivateKey).pubBytes())
+      prefix + bytesToHex(nobleEd25519.getPublicKey(rawPrivateKey))
     return { privateKey, publicKey }
   },
 
-  sign(message, privateKey): string {
-    // caution: Ed25519.sign interprets all strings as hex, stripping
-    // any non-hex characters without warning
-    if (!Array.isArray(message)) {
-      throw new Error('message must be array of octets')
-    }
+  sign(message: ByteArray, privateKey: HexString): string {
+    assert.ok(
+      Array.isArray(message) || message instanceof Uint8Array,
+      'message must be array of octets',
+    )
+    assert.ok(
+      privateKey.length === 66,
+      'private key must be 33 bytes including prefix',
+    )
     return bytesToHex(
-      Ed25519.sign(message, hexToBytes(privateKey).slice(1)).toBytes(),
+      nobleEd25519.sign(new Uint8Array(message), privateKey.slice(2)),
     )
   },
 
-  verify(message, signature, publicKey): boolean {
-    return Ed25519.verify(
-      message,
-      hexToBytes(signature),
-      hexToBytes(publicKey).slice(1),
+  verify(
+    message: ByteArray,
+    signature: HexString | Uint8Array,
+    publicKey: string,
+  ): boolean {
+    return nobleEd25519.verify(
+      signature,
+      new Uint8Array(message),
+      publicKey.slice(2),
     )
   },
 }
 
-function select(algorithm): any {
+function select(algorithm: Algorithm) {
```

Comment:
This function should have a return type

Reply:
added SigningMethod interface to match existing code `method = select(...`)